microbench/src/main/java/io/netty/microbenchmark/common/IsValidIpV4Benchmark.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbenchmark.common;
+
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.concurrent.TimeUnit;
+
+@Threads(1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 3)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+public class IsValidIpV4Benchmark extends AbstractMicrobenchmark {
+
+    @Param({ "127.0.0.1", "255.255.255.255", "1.1.1.1", "127.0.0.256", "127.0.0.1.1", "127.0.0", "[2001::1]" })
+    private String ip;
+
+    // Old
+
+    public static boolean isValidIpV4Address_old(String value) {
+
+        int periods = 0;
+        int i;
+        int length = value.length();
+
+        if (length > 15) {
+            return false;
+        }
+        char c;
+        StringBuilder word = new StringBuilder();
+        for (i = 0; i < length; i ++) {
+            c = value.charAt(i);
+            if (c == '.') {
+                periods ++;
+                if (periods > 3) {
+                    return false;
+                }
+                if (word.length() == 0) {
+                    return false;
+                }
+                if (Integer.parseInt(word.toString()) > 255) {
+                    return false;
+                }
+                word.delete(0, word.length());
+            } else if (!Character.isDigit(c)) {
+                return false;
+            } else {
+                if (word.length() > 2) {
+                    return false;
+                }
+                word.append(c);
+            }
+        }
+
+        if (word.length() == 0 || Integer.parseInt(word.toString()) > 255) {
+            return false;
+        }
+
+        return periods == 3;
+    }
+
+    // New
+
+    private static boolean isValidIpV4Word(CharSequence word, int start, int endExclusive) {
+        int len = endExclusive - start;
+        char c0, c1, c2;
+        if (len < 1 || len > 3 || (c0 = word.charAt(start)) < '0') {
+            return false;
+        }
+        if (len == 3) {
+            return (c1 = word.charAt(start + 1)) >= '0' &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (9) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)